WordPress/src/main/java/org/wordpress/android/ui/domains/DomainsDashboardViewModel.kt

```diff
@@ -102,32 +98,32 @@ class DomainsDashboardViewModel @Inject constructor(
     private fun buildDashboardItems(site: SiteModel, plans: List<PlanModel>, domains: List<Domain>) {
         val listItems = mutableListOf<DomainsDashboardItem>()
 
+        listItems += SiteDomainsHeader(UiStringRes(R.string.domains_free_domain))
+
         val freeDomain = domains.firstOrNull { it.wpcomDomain }
         val freeDomainUrl = freeDomain?.domain ?: getCleanUrl(site.unmappedUrl)
         val freeDomainIsPrimary = freeDomain?.primaryDomain ?: false
 
-        listItems += FreeDomain(UiStringText(freeDomainUrl), freeDomainIsPrimary, this::onChangeSiteClick)
+        listItems += SiteDomains(
+            UiStringText(freeDomainUrl),
+            UiStringRes(R.string.domains_site_domain_never_expires),
+            freeDomainIsPrimary
+        )
 
         val customDomains = domains.filter { !it.wpcomDomain }
         val hasCustomDomains = customDomains.isNotEmpty()
         val hasDomainCredit = isDomainCreditAvailable(plans)
         val hasPaidPlan = !SiteUtils.onFreePlan(site)
 
-        listItems += buildCustomDomainItems(customDomains, hasCustomDomains)
-
-        listItems += buildCtaItems(freeDomainUrl, hasCustomDomains, hasDomainCredit, hasPaidPlan)
+        listItems += buildCustomDomainItems(site, customDomains)
```

Comment:
This needs to be wrapped in a check for hasCustomDomains.  Otherwise there'll be a dangling **Other domains for ...** header when there are no custom domains.

Reply:
Applied the check before adding the custom domain items to the list cb0bf7c3f86331c785f5136243d6e189c254cae7

Reply:
@ravishanker, actually, that was the intention. In the design, the "Other domains for ..." title is visible even there are no custom domains. It makes sense to me because that section is meant for other domains, and we display the domain purchasing card if there are no domains under the "Other domains" section.
If you agree, I would like to revert your commit. Otherwise, we can consult with the designer here and also inform the iOS team to align with this decision.